checker/src/main/java/org/checkerframework/checker/nullness/collection-object-parameters-may-be-null.astub

```diff
@@ -0,0 +1,254 @@
+//     "Attempts to insert a null key will throw NullPointerException. Attempts to test for the
+//     presence of a null key or to remove one will, however, function properly. Null values are
+//     permitted."
+//
+// Here are some exceptions: methods whose specification is ambiguous (says "such that o.equals(e)"
+// or "such that Objects.equals(o, e)") but can be interpreted to say that null is a permitted
+// value:
+//
+//   ArrayBlockingQueue
+//   ArrayDeque
+//   ConcurrentLinkedDeque
+//   ConcurrentLinkedQueue
+//   CopyOnWriteArrayList
+//   CopyOnWriteArraySet
+//   LinkedBlockingDeque
+//   LinkedBlockingQueue
+//   LinkedTransferQueue
+//   PriorityBlockingQueue
+//   PriorityQueue
+//     contains
+//     remove
+//
+//   ArrayDeque
+//     removeFirstOccurrence
+//     removeLastOccurrence
+//
+//   Collections
+//     frequency
+//
+// Special cases:
+//   EnumMap:
+//     "Attempts to insert a null key will throw NullPointerException. Attempts to test for the
+//     presence of a null key or to remove one will, however, function properly. Null values are
+//     permitted."
+//
+// Many additional method implementations are null-safe, but the specification does not guarantee
+// that all overriding implementations in subclasses will be.
+
+
+package java.util;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class AbstractCollection {
+    @Pure public boolean contains(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+    @Pure public boolean containsAll(Collection<?> c);
+    public boolean removeAll(Collection<?> c);
+    public boolean retainAll(Collection<?> c);
+}
+
+public abstract class AbstractList<E> extends AbstractCollection<E> implements List<E> {
+    public int indexOf(@Nullable Object o);
+    public int lastIndexOf(@Nullable Object o);
+}
+
+class AbstractMap {
+    public boolean containsValue(@Nullable Object value);
+    public boolean containsKey(@Nullable Object key);
+    public @Nullable V get(@Nullable Object key);
+    public @Nullable V remove(@Nullable Object key);
+}
+
+class AbstractSet {
+  public boolean removeAll(Collection<?> c);
+}
+
+public class ArrayList<E> extends AbstractList<E>
+        implements List<E>, RandomAccess, Cloneable, java.io.Serializable
+{
+    public boolean contains(@Nullable Object o);
+    public int indexOf(@Nullable Object o);
+    public int lastIndexOf(@Nullable Object o);
+    public boolean remove(@Nullable Object o);
+}
+
+public interface Collection<E> extends Iterable<E> {
+    boolean contains(@Nullable Object o);
+    boolean remove(@Nullable Object o);
+    boolean containsAll(Collection<? extends @Nullable Object> c);
+    boolean addAll(Collection<? extends @Nullable E> c);
+    boolean retainAll(Collection<? extends @Nullable Object> c);
+}
+
+public interface Comparator<T> {
+    int compare(@Nullable T o1, @Nullable T o2);
+
+}
+
+class ConcurrentSkipListSet {
+    public boolean removeAll(Collection<?> c);
```

Comment:
I believe this should forbid nulls.

Reply:
That would be the same as the annotation in the JDK.
This is trying to be more lenient and permit collections of nullable type.

Are you proposing that the stub file in general should not relax the signature of `removeAll`?

Reply:
Right, I'm proposing to match the annotated JDK. `ConcurrentSkipListSet`, even with a `Comparator` that permits `null`, has documentation on methods like `remove` and `removeAll` to suggest that they reject `null`. This is in contrast to `TreeSet`, which allows null queries if its `Comparator` does.

Reply:
Thanks.  This comment only applies to ConcurrentSkipList and not to other methods, so I will only change ConcurrentSkipList by removing these lines from the stub file.